lib/ansible/modules/network/ios/ios_logging.py

```diff
@@ -0,0 +1,352 @@
+#
+# This file is part of Ansible by Red Hat
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
+
+DOCUMENTATION = """
+---
+module: ios_logging
+version_added: "2.4"
+author: "Trishna Guha (@trishnag)"
+short_description: Manage logging on network devices
+description:
+  - This module provides declarative management of logging
+    on Cisco Ios devices.
+options:
+  dest:
+    description:
+      - Destination of the logs.
+    choices: ['on', 'host', console', 'monitor', 'buffered']
+  name:
+    description:
+      - If value of C(dest) is I(file) it indicates file-name,
+        for I(user) it indicates username and for I(host) indicates
+        the host name to be notified.
+  size:
+    description:
+      - Size of buffer. The acceptable value is in range from 4096 to
+        4294967295 bytes.
+  facility:
+    description:
+      - Set logging facility.
+  level:
+    description:
+      - Set logging severity levels.
+  collection:
+    description: List of logging definitions.
+  purge:
+    description:
+      - Purge logging not defined in the collections parameter.
+    default: no
+  state:
+    description:
+      - State of the logging configuration.
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: configure host logging
+  ios_logging:
+    dest: host
+    name: 172.16.0.1
+    state: present
+- name: remove host logging configuration
+  ios_logging:
+    dest: host
+    name: 172.16.0.1
+    state: absent
+- name: configure console logging level and facility
+  ios_logging:
+    dest: console
+    facility: local7
+    level: debugging
+    state: present
+- name: enable logging to all
+  ios_logging:
+    dest : on
+- name: configure buffer size
+  ios_logging:
+    dest: buffered
+    size: 5000
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device
+  returned: always
+  type: list
+  sample:
+    - logging facility local7
+    - logging host 172.16.0.1
+"""
+
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ios import get_config, load_config
+from ansible.module_utils.ios import ios_argument_spec, check_args
+
+
+def validate_size(value, module):
```

Comment:
Seems like this function is not called from within the module.
You are missing the validate function that calls validation on each param.
Reference: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_vlan.py#L131-L137
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_vlan.py#L175